sci-libs/openblas/openblas-0.3.6.ebuild

```diff
@@ -11,34 +11,45 @@ SRC_URI="https://github.com/xianyi/OpenBLAS/tarball/v${PV} -> ${P}.tar.gz"
 LICENSE="BSD"
 SLOT="0"
 KEYWORDS="~amd64 ~x86 ~amd64-linux ~x86-linux ~x64-macos ~x86-macos"
-IUSE="dynamic openmp pthread serial static-libs eselect-ldso index-64bit"
-REQUIRED_USE="?? ( openmp pthread serial )"
+IUSE="dynamic eselect-ldso index-64bit openmp pthread"
+REQUIRED_USE="?? ( openmp pthread )"
 
 RDEPEND="
-eselect-ldso? ( >=app-eselect/eselect-blas-0.2
-				!app-eselect/eselect-cblas
-				>=app-eselect/eselect-lapack-0.2 )
-"
+	eselect-ldso? ( >=app-eselect/eselect-blas-0.2
+			!app-eselect/eselect-cblas
+			>=app-eselect/eselect-lapack-0.2 )"
+
 DEPEND="virtual/pkgconfig"
 
 PATCHES=( "${FILESDIR}/shared-blas-lapack.patch" )
 
-openblas_flags() {
-	local flags=()
+src_prepare() {
+	default
+	# Set compiler and common CFLAGS.
+	sed \
+		-e "s:^#\s*\(CC\)\s*=.*:\1=$(tc-getCC):" \
+		-e "s:^#\s*\(FC\)\s*=.*:\1=$(tc-getFC):" \
+		-e "s:^#\s*\(COMMON_OPT\)\s*=.*:\1=${CFLAGS}:" \
+		-i "$S"/Makefile.rule || die
+}
```

Comment:
Could this be simplified to `/^#\s*CC/cCC=$(tc-getCC)`?

Reply:
```
sed \
                -e "s:^#\s*\(CC\)\s*=.*:\1=$(tc-getCC):" \
                -e "s:^#\s*\(FC\)\s*=.*:\1=$(tc-getFC):" \
                -e "s:^#\s*\(COMMON_OPT\)\s*=.*:\1=${CFLAGS}:" \
                -i "$S"/Makefile.rule || die
}
```

I think this code block already simplified. Also it will be easy to read code for the next maintainer of the package. But if you believe this is the missing part of ebuild feel free to update it. It is your project.

I also didn't agreed with you about unnecessary REQUIRED_USE="?? ( openmp pthread  )"
But implemented for your wish.

Also if you don't want to work together please inform me i can close PR. No problem for me.  

```

Reply:
Hello Hasan,

> ```
> sed \
>                 -e "s:^#\s*\(CC\)\s*=.*:\1=$(tc-getCC):" \
>                 -e "s:^#\s*\(FC\)\s*=.*:\1=$(tc-getFC):" \
>                 -e "s:^#\s*\(COMMON_OPT\)\s*=.*:\1=${CFLAGS}:" \
>                 -i "$S"/Makefile.rule || die
> }
> ```
> 
> I think this code block already simplified. Also it will be easy to read code for the next maintainer of the package. But if you believe this is the missing part of ebuild feel free to update it. It is your project in the future.

There is no point to capture a fixed string `\(CC\)` and replay it with `\1`.   What you want is to find the line and replace it with what you want.

I point this out to you, in hope that you could understand the reason and take the recommendation in your next pull request.

> I also didn't agreed with you about unnecessary REQUIRED_USE="?? ( openmp pthread )"
> But implemented for your wish.

Thanks.

If you can't understand, ask.

> Also if you don't want to work together please inform me i can close PR. No problem for me.

I don't know how you got the feeling that I do not want to work with you. I hope you could concentrate on technical arguments. 

I am not playing hard on you.  Learning how to convey your technical idea is a deliberate process.  Your idea that seems trivial might not apply well to others.

Have a nice day,
Benda

Reply:
Thank you for your regular expression lessons Benda.

But sometimes seeing things simple like checking is there any active PR before opening a new one is more important i think.

Have a nice day,
Hasan